dandiapi/api/management/commands/import_dandisets.py

```diff
@@ -69,23 +133,33 @@ def import_dandiset_from_response(
         f'Importing dandiset {identifier} "{dandiset_api_response["draft_version"]["name"]}" as dandiset {dandiset.identifier}'  # noqa: E501
     )
 
-    import_versions_from_response(api_url, versions, dandiset)
+    import_versions_from_response(api_url, versions, dandiset, include_assets)
 
 
 @transaction.atomic
 def import_dandisets_from_response(
-    api_url: str, dandiset_api_response: dict, identifier_offset: int, should_replace: bool
+    api_url: str,
+    dandiset_api_response: dict,
+    identifier_offset: int,
+    should_replace: bool,
+    include_assets: bool,
 ):
     """Import dandisets given a response from /api/dandisets/."""
     for result in dandiset_api_response['results']:
         import_dandiset_from_response(
-            api_url, result, identifier_offset=identifier_offset, should_replace=should_replace
+            api_url,
+            result,
+            identifier_offset=identifier_offset,
+            should_replace=should_replace,
+            include_assets=include_assets,
         )
 
     # Handle API pagination
     if dandiset_api_response.get('next'):
         new_dandisets = requests.get(dandiset_api_response.get('next')).json()
-        import_dandisets_from_response(api_url, new_dandisets, identifier_offset, should_replace)
+        import_dandisets_from_response(
+            api_url, new_dandisets, identifier_offset, should_replace, include_assets
+        )
```

Comment:
Same comment here regarding recursion (for the record happy to have a discussion about this, I just don't see a huge reason to use recursion here).

Reply:
For some reason, recursion felt more natural to me for this. But I think you're right that iteration is more practical here, especially consider we'll be running this on Heroku dynos with limited memory. I'll update it to use iteration